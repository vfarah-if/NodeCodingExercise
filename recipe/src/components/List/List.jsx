import React, { useState } from 'react';
import './style/index.css';

const List = () => {
	const [items, setItems] = useState([]);
	const [newItem, setNewItem] = useState('');
	const submitHandler = (event) => {
        setItems(items.concat([newItem]));
        setNewItem('');
        event.preventDefault();
	};

	return (
		<div className="todo-list">
			<h2>TODO List</h2>
			<ul>
				{items.map((item) => (
					<li key={item}>{item}</li>
				))}
			</ul>
			<form onSubmit={submitHandler} data-testid="form"> 
				<div className="form-group">
					<input
                        id="newItem"
						className="form-control"
						type="text"
						value={newItem}
                        onChange={(e) => setNewItem(e.target.value)}
                        placeholder="Enter new item"
					/>
				</div>
                <input type="submit" value="Add" className="btn btn-primary"></input>
			</form>
		</div>
	);
};

export default List;
