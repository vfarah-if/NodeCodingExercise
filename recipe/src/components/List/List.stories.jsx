import React, { Fragment } from 'react';
import List from './List';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
	title: 'Component/List',
	Component: List,
};

export const DefaultListStory = () => (
	<Fragment>
		<List />
	</Fragment>
);
