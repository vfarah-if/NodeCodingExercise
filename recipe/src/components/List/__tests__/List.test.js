import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import List from '../List';

describe('List', () => {
	const item1 = 'TODO 1: Add the first item';
	const item2 = 'TODO 2: Add the second item';
	
	test('should add one item into the list', async () => {
		const component = render(<List />);
		const input = await component.findByPlaceholderText('Enter new item');
		editValue(input, item1);

		await submitAddButton(component);

		const listItem = await component.findByText(item1);
		expect(listItem).toMatchSnapshot();	
    });
    
    test('should clear the list after adding another item', async () => {
		const component = render(<List />);
		const input = await component.findByPlaceholderText('Enter new item');		
		
		editValue(input, item1);		
		await submitAddButton(component);
		expect(input.value).toBe('');

		editValue(input, item2);	
		await submitAddButton(component);
		expect(input.value).toBe('');

		const listItem = await component.findByText(item2);
		expect(listItem).toMatchSnapshot();	
	});
});

async function submitAddButton(component) {
	const form = await component.findByTestId('form');
	expect(form).toBeDefined();
	fireEvent.submit(form);
}

function editValue(input, item1) {
	expect(input).toBeDefined();
	fireEvent.change(input, {
		target: { value: item1 },
	});
	expect(input.value).toBe(item1);
}

