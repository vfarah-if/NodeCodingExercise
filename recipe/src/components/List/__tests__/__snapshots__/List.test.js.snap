// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should add one item into the list 1`] = `
<li>
  TODO 1: Add the first item
</li>
`;

exports[`List should clear the list after adding another item 1`] = `
<li>
  TODO 2: Add the second item
</li>
`;
